app/code/Magento/CatalogSearch/Model/Adapter/Mysql/Aggregation/DataProvider/SelectBuilderForAttribute.php

```diff
@@ -0,0 +1,100 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+namespace Magento\CatalogSearch\Model\Adapter\Mysql\Aggregation\DataProvider;
+
+use Magento\CatalogSearch\Model\Adapter\Mysql\Aggregation\DataProvider\SelectBuilderForAttribute\StockConditionJoiner;
+use Magento\Customer\Model\Session;
+use Magento\Eav\Model\Entity\Attribute\AbstractAttribute;
+use Magento\Framework\App\ResourceConnection;
+use Magento\Framework\App\ScopeResolverInterface;
+use Magento\Framework\DB\Select;
+use Magento\Framework\Search\Request\BucketInterface;
+use Magento\Store\Model\Store;
+
+/**
+ * Build select for attribute.
+ */
+class SelectBuilderForAttribute
+{
+    /**
+     * @var ScopeResolverInterface
+     */
+    private $scopeResolver;
+
+    /**
+     * @var ResourceConnection
+     */
+    private $resource;
+
+    /**
+     * @var StockConditionJoiner
+     */
+    private $stockConditionJoiner;
+
+    /**
+     * @var Session
+     */
+    private $customerSession;
+
+    /**
+     * @param ResourceConnection $resource
+     * @param ScopeResolverInterface $scopeResolver
+     * @param StockConditionJoiner $stockConditionJoiner
+     * @param Session $customerSession
+     */
+    public function __construct(
+        ResourceConnection $resource,
+        ScopeResolverInterface $scopeResolver,
+        StockConditionJoiner $stockConditionJoiner,
+        Session $customerSession
+    ) {
+        $this->resource = $resource;
+        $this->scopeResolver = $scopeResolver;
+        $this->stockConditionJoiner = $stockConditionJoiner;
+        $this->customerSession = $customerSession;
+    }
+
+    /**
+     * @param Select $select
+     * @param AbstractAttribute $attribute
+     * @param int $currentScope
+     *
+     * @return Select
+     */
+    public function execute(Select $select, AbstractAttribute $attribute, int $currentScope): Select
```

Comment:
why does builder has method execute, but not build?

Reply:
Fixed.